libs/api/domains/user-profile/src/lib/userProfile.service.ts

```diff
@@ -49,6 +52,25 @@ export class UserProfileService {
     return this.userProfileApi.withMiddleware(new AuthMiddleware(auth))
   }
 
+  private async handleRawRes<T>(
+    promise: Promise<ApiResponse<T>>,
+  ): Promise<T | null> {
+    return promise.then(
+      (response) => {
+        if (response.raw.status === IGNORED_STATUS) {
+          return null
+        }
+        return response.value()
+      },
+      (error) => {
+        if (error instanceof FetchError && error.status === IGNORED_STATUS) {
+          return null
+        }
+        throw error
+      },
+    )
+  }
```

Comment:
This is a nice utility wrapper to handle our 204 responses. Can you move it to the `@island.is/clients/middlwares` lib as the `handle404` handler.